livekit-rtc/livekit/rtc/participant.py

```diff
@@ -292,7 +292,15 @@ async def set_attributes(self, attributes: dict[str, str]) -> None:
         """
         req = proto_ffi.FfiRequest()
         req.set_local_attributes.local_participant_handle = self._ffi_handle.handle
-        req.set_local_attributes.attributes.update(attributes)
+        existing_attributes = {
+            entry.key: entry.value for entry in req.set_local_attributes.attributes
+        }
+        existing_attributes.update(attributes)
+
+        for key, value in existing_attributes.items():
+            entry = req.set_local_attributes.attributes.add()
+            entry.key = key
+            entry.value = value
 
```

Comment:
Was the logic wrong before?

Reply:
it showed up as a type error where lists and dicts were mixed

Reply:
but this might also be as a result of how the `map` type in proto get's interpreted in proto2 vs proto3 syntax, not sure